### ISSUE TITLE
Crash when using a malformed --lua-version

### ISSUE THREAD

**Issue by kalekje** [luarocks/luarocks#1252]
I use Lua for LuaLaTeX, and am attempting to get luarocks working. I had the following error come up and it mentioned to report it:
```
C:\Users\Kale>luarocks config lua_ --lua-version "C:\Program Files\MiKTeX 2.9\miktex\bin\x64\texlua.exe"

Error: LuaRocks 3.4.0 bug (please report at https://github.com/luarocks/luarocks/issues).
Arch.: win32-x86_64 (fs_use_modules = true) (binary)
[string "src/luarocks/deps.lua"]:656: bad argument #1 to 'floor' (number expected, got nil)
stack traceback:
        [C]: in function 'math.floor'
        [string "src/luarocks/deps.lua"]:656: in upvalue 'lua_h_exists'
        [string "src/luarocks/deps.lua"]:684: in upvalue 'find_lua_incdir'
        [string "src/luarocks/deps.lua"]:701: in function 'luarocks.deps.check_lua_incdir'
        [string "src/luarocks/cmd/config.lua"]:243: in function 'luarocks.cmd.config.command'
        (...tail calls...)
        [C]: in function 'xpcall'
        [string "src/luarocks/cmd.lua"]:661: in function 'luarocks.cmd.run_command'
        [string "luar
```

**Comment by daurnimator** [comment]
Could you share your lua configuration (you could use e.g. `luarocks config`)? Looking at the code, this would appear to happen if your `lua_version` was configured to be a non-number value.

**Comment by Zash** [comment]
`--lua-version "C:\..."` does it, same happens if you `--lua-version=lua5.4`

The correct thing is to `--lua-version=5.4`, i.e. **only** the version number.

**Comment by hishamhm** [comment]
@Zash thank you for spotting that! Fixed in 99b1ba350d3c933322db541f86ee7a5f70995e30

**Comment by hishamhm** [comment]
(The issue most likely persists if one uses `luarocks config`, but I suppose there's lots of other config keys that don't do thorough validation of their formats...)
